test/has_attributes_test.rb

```diff
@@ -0,0 +1,88 @@
+require "test_helper"
+require "openxml/has_attributes"
+
+class HasAttributesTest < Minitest::Test
+  context "ClassMethod.attribute" do
+    should "define a reader method for the attribute" do
+      element = class_with_attribute :an_attribute
+      assert element.new.respond_to? :an_attribute
+    end
+
+    should "define a writer method for the attribute" do
+      element = class_with_attribute :an_attribute
+      assert element.new.respond_to? :an_attribute=
+    end
+
+    should "be named by its camelCased version if no displays_as given" do
+      element = class_with_attribute :an_attribute
+      assert_equal :anAttribute, element.attributes[:an_attribute][0]
+    end
+
+    should "be named by its displays_as property if given" do
+      element = class_with_attribute :an_attribute, displays_as: :theAttr
+      assert_equal :theAttr, element.attributes[:an_attribute][0]
+    end
+
+    context "in a with_namespace block" do
+      should "use the provided namespace" do
+        element = Class.new {
+          include OpenXml::HasAttributes
+          with_namespace :a_namespace do
+            attribute :an_attribute
+          end
+        }
```

Comment:
🎨 The should probably be `do...end` style, right? — same logic as `.tap`: the interior is definitely mutating the class; the block is for its side-effects not its return value.